src/Aspire.Dashboard/Model/ContainerViewModel.cs

```diff
@@ -23,4 +23,16 @@ internal override bool MatchesFilter(string filter)
     {
         return base.MatchesFilter(filter) || Image.Contains(filter, StringComparisons.UserTextSearch);
     }
+
+    public override bool Equals(ResourceViewModel? other)
+    {
+        return other is ContainerViewModel container
```

Comment:
I had the same thought. The benefits are obvious, but this feels like a maintenance headache. A test like James mentioned could work but yeah wouldn't be quick to write. I think in WebTools we treated a similar scenario as if it were a public api, so just ensured that a test would fail if the public api was changed (added or removed). Slightly easier to write, more annoying to maintain than the reflection way.

Reply:
I'll add a comment. The type small and the code is adjacent.

This might not even end up staying. This code will churn a bunch over the next week or so. I'm trying to peel off small PRs to make reviews of larger work easier.

Reply:
Added comments.